### DISCUSSION TITLE
Build NextJS application in development mode

### DISCUSSION THREAD

**Discussion by saivanji** [vercel/next.js#45262] [Help]
How can I build the application in development mode so I can see the original text of the errors in the console?

I've tried `NODE_ENV=development next build` but it still builds the application in production mode and errors are minified.

The reason for that is that we have hydration errors in our website which are not appearing in development mode, which make it very tricky to debug.

Having the application bundled in the development mode will let us see the root cause of the markup mismatch (HTML from server VS client HTML)

**Comment by icyJoseph** [comment]
This is very tricky I am afraid... the depth of NODE_ENV is hard to deal with, because it crosses even through node_modules, and invokes different paths of code, and certain code elimination techniques kick in as well. 

I think you might be able to override it through webpack, but I am really not sure that'll go anywhere. Not even gonna try.

For your case, perhaps run the site with JS disabled, or query it from a non-browser client, and then try to diff that with what the browser paints? Use incognito mode to prevent extensions code from affecting the diff. 

It is also possible to place breakpoints in the browser, by opening the call stack printed by React, and try to follow where does the difference come from. I've done this about 3 times, only worked in 2, the last time, it was impossible to figure out where in the tree things went south.

**Comment by guitheengineer** [comment]
You're able to build in development mode with `NODE_ENV=development next start"` and disable minification with `swcMinify: false` on your next.config, but even then it won't show explicitly your hydration errors. Is better to just run the application locally and debug the error from there.
